packages/frontend-2/components/workspace/header/Header.vue

```diff
@@ -173,4 +193,28 @@ const onActionChosen = (params: { item: LayoutMenuItem; event: MouseEvent }) =>
       break
   }
 }
+
+const descriptionRef = ref<HTMLElement | null>(null)
+const hasOverflow = ref(false)
+
+const checkOverflow = () => {
+  if (descriptionRef.value) {
+    hasOverflow.value =
+      descriptionRef.value.scrollHeight > descriptionRef.value.clientHeight
+  }
+}
+
+const isDescriptionDialogOpen = ref(false)
+
+const showDescriptionDialog = () => {
+  isDescriptionDialogOpen.value = true
+}
+
+watch(
+  () => props.workspaceInfo,
+  () => {
+    nextTick(checkOverflow)
```

Comment:
This is still imperative and still unreliable - whether or not the DOM node will be ready on next tick could change, and there's also the fact that window resizes should retrigger size calculations as well.

You should be using some kind of reactive vueuse DOM size tracking util instead, so that hasOverflow can be a declarative computed and you don't even have to worry about when and how to trigger recalculation

Reply:
Thanks. There is no vueuse util that can read scroll height or client height. At least one that I could find. Any suggestions?

Reply:
@andrewwallacespeckle What about useElementSize or useElementBounding? Even if it doesn't give you those size values you're after, as long as it updates when the element size updates, you can at the very least retrigger overflow calculation when that happens (e.g. with a watcher)

Reply:
Thanks for your help on this! `useElementSize` worked perfectly